R/join_key.R

```diff
@@ -2,30 +2,35 @@
 #'
 #' @description `r lifecycle::badge("stable")`
 #'
-#' @details `join_key()` will create a relationship for the variables on a pair
-#' of datasets.
+#' @description `Create a relationship for the variables on a pair of datasets.
+#' It is directed by default with `dataset_1` being the parent.
 #'
-#' @param dataset_1,dataset_2 (`character(1)`) dataset names. If `dataset_2` is omitted,
-#'  a primary key for `dataset_1` is created.
-#' @param keys (optionally named `character`) where `names(keys)` are columns in `dataset_1`
-#' corresponding to columns of `dataset_2` given by the elements of `keys`.
+#' @param dataset_1,dataset_2 (`character(1)`) Dataset names. When `dataset_2` is omitted,
+#' a primary key for `dataset_1` is created.
+#' @param keys (optionally named `character`) Column mapping between the datasets,
+#' where `names(keys)` maps columns in `dataset_1` corresponding to columns of
+#' `dataset_2` given by the elements of `keys`.
 #'
 #' If unnamed, the same column names are used for both datasets.
 #'
 #' If any element of the `keys` vector is empty with a non-empty name, then the name is
 #' used for both datasets.
+#' @param parent (`character(1)`) Indicates the parent dataset in a parent-child
+#' relationship or `none` if it is an undirected relationship.
+#' One of `dataset_1`, `dataset_2` or `none`.
 #'
 #' @return object of class `join_key_set` to be passed into `join_keys` function.
 #'
-#' @seealso [join_keys()]
+#' @seealso [join_keys()], [parents()]
 #'
 #' @export
 #'
 #' @examples
 #' join_key("d1", "d2", c("A"))
 #' join_key("d1", "d2", c("A" = "B"))
 #' join_key("d1", "d2", c("A" = "B", "C"))
-join_key <- function(dataset_1, dataset_2 = dataset_1, keys) {
+join_key <- function(dataset_1, dataset_2 = dataset_1, keys, parent = c("dataset_1", "dataset_2", "none")) {
+  parent <- match.arg(parent)
```

Comment:
I find it little odd with proposed way of specifying parent. `parent` is not exactly a parent but it rather sets the direction of the relationship between `dataset_1` and `dataset_2`. Here, information in `parent` is redundant and could be easily replaced by inverting `dataset_1` with `dataset_2`. I think there is too many (two :D) ways to achieve the same result. I propose

```suggestion
join_key <- function(dataset_1, dataset_2 = dataset_1, keys, directed = TRUE) {
  checkmate::assert_flag(directed)
```

And instead of doing
```r
# setting B as parent
join_key(dataset1 = "A", dataset_2 = "B", parent = "dataset_2")
join_key(dataset1 = "B", dataset_2 = "A") 

# setting no parent
join_key(dataset1 = "A", dataset_2 = "B", parent = "none")
```

I would prefer only

```r
# setting B as parent 
join_key(dataset1 = "B", dataset_2 = "A")

# setting no parent
join_key(dataset1 = "A", dataset_2 = "B", directed = FALSE)
```

Reply:
I like the explicit usage of indicating if it's one or the other, but given that this will hardly ever be used I don't oppose the flag. Especially as this avoids the issue of different parameters values.

I'll make the changes, which I believe will resolve all the comments from your review.